services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/volumes.py

```diff
@@ -85,7 +103,7 @@ def target(cls, path: Path) -> str:
         return f"{target_path}"
 
     @classmethod
-    def _volume_name(cls, path: Path) -> str:
+    def volume_name(cls, path: Path) -> str:
         return f"{path}".replace(os.sep, "_")
```

Comment:
ah, this is where the `_` comes from, or occurs again. 

And since I have worked with simcore already a bit it is dawning on me why this `_` subsitution is done: I guess docker named volumes dont support the `/` character, so we switch with `_`. This is a bad hack though, since paths can contain `_` in foldernames/filenames, so this is a **lossy** mapping that cant strictly be reverted.

I would at least add clear explanations, or reconsider... and maybe a unittest what happens when a folder contains an underscore?

Reply:
but maybe I worry too much, let me know if this is an issue at all

Reply:
Here I just reuse what is there so I do not know the details of that, please ask @GitHK

Reply:
So there is a reason why the name is private.

The following considerations apply here. There is no from `volume_name` to `path` requirement. Since it's not required and there is no reason to have such transformation this is simply enugh.

@matusdrobuliak66 since you are exposing this please add a docstring to it and maybe point out that it is not possible to go back to the orignal path form the volume name